sdk/cmake/CMakeMacros.cmake

```diff
@@ -533,7 +533,8 @@ function(add_importlibs _module)
     foreach(LIB ${ARGN})
         if("${LIB}" MATCHES "msvcrt")
             target_compile_definitions(${_module} PRIVATE _DLL __USE_CRTIMP)
-            target_link_libraries(${_module} msvcrtex)
+            # set_module_type might not have been called at this point, so use a generator expression
+            target_link_libraries(${_module} "$<$<IN_LIST:$<TARGET_PROPERTY:${_module},REACTOS_MODULE_TYPE>,win32gui;win32cui;win32dll;win32ocx;cpl>:msvcrtex>")
```

Comment:
Can't we do ```target_link_libraries(msvcrt INTERFACE msvcrtex)``` and get all of this automatically?

Reply:
Nope; this would link msvcrtex after libmsvcrt, and we want it to be before...

Reply:
Actually, we can, with a bit of trickery :-)

Reply:
If we can mark everything that comes from msvcrtex, that is also in msvcrt as private, the order shouldn't matter.

Reply:
True, but then you have to link msvcrtex with libmsvcrt anyway, so you end up with a libmsvcrt msvcrtex libmsvcrt linking order